compiler/rustc_parse/src/validate_attr.rs

```diff
@@ -149,7 +149,9 @@ pub fn check_builtin_meta_item(
 ) {
     // Some special attributes like `cfg` must be checked
     // before the generic check, so we skip them here.
-    let should_skip = |name| name == sym::cfg;
+    // Coverage attributes are checked by the `check_attr` pass, which gives
+    // better error messages.
+    let should_skip = |name| name == sym::cfg || name == sym::coverage;
```

Comment:
this feels undesirable to me 🤔 I feel like skipping attributes here can easily result in accidentally not checking something if the builtin attr definition ever changes.

I feel like if the error message isn't good enough with the automatic meta item check, that check should be improved/customizeable instead of disabling it and writing it by hand.

Reply:
That's fair.

Currently the syntax is still checked in `check_attr`, and is covered by tests, but I can look into what it would take to give good checking/errors directly in the parse layer.